packages/query-persist-client-core/src/persist.ts

```diff
@@ -52,6 +53,20 @@ export interface PersistQueryClientOptions
     PersistedQueryClientSaveOptions,
     PersistQueryClienRootOptions {}
 
+/**
+ * Checks if emitted event is about cache change and not about observers.
+ * Useful for persist, where we only want to trigger save when cache is changed.
+ */
+function isCacheableEventType(eventType: NotifyEventType) {
```

Comment:
@TkDodo I placed this function here as in internal one for now. Let me know if there is a better place for that.

Reply:
I think we should check that Dominik is happy with [this idea](https://github.com/TanStack/query/pull/4884#discussion_r1088716259), but I think we could start on the tests for the new behaviour if you're up for it 🙂

Reply:
sure I am. Although no idea where to start

Reply:
It looks like the relevant tests are here: 

https://github.com/TanStack/query/blob/main/packages/query-persist-client-core/src/__tests__/persist.test.ts#L22

I think this is probably going to be non-trivial to test. I'd be happy to take a look but I won't have time for a little while, although I think Dominik said he'd be willing to lend a hand 🤞

Reply:
awesome, thanks!